### ISSUE TITLE
Role filter dropdown, per role SR graph 

### ISSUE THREAD

**Issue by synap5e** [overtrack-gg/overtrack-web#68]
With role queue, SR tracks separately for each role. Users should be able to filter their games by role. This could be implemented by a second dropdown next to season selection (although filtering would be clientside). 
We could remove the "games" title to make it fit
![image](https://user-images.githubusercontent.com/2515062/62013597-e0968280-b1e8-11e9-9af0-59c8d9c8d018.png)

Games from the v2 `/overwatch/games` API have a new field `role` that can be `tank`, `damage`, or `support`. This will need to be added to the game model.

**New dropdown behavior:**
 - appears in seasons where the games have the `role` field
 - defaults to "all roles" if no setting in localstorage
 - saves selection to localstorage
 - filters (clientside, no refetch) games to only that role (or all roles if that's selected)
 - Sets mini SR graph to show SR for selected role

**New SR graph behavior**
 - shows SR only of selected role
 - if "all roles" is selected, shows SR graph of the role of the most recent game

**New "Current SR" behavior**
 - shows SR only of selected role
 - if "all roles" is selected, shows SR of the role of the most recent game
 - Has the icon of the most recent role

**Comment by JWSenteney** [comment]
The Games list part of this task is done, and ready for a PR.

<img width="1126" alt="Screen Shot 2019-08-01 at 4 07 42 PM" src="https://user-images.githubusercontent.com/1554771/62324340-46983980-b477-11e9-8b85-b735db9e98cc.png">

Instead of a dropdown for selecting roles, I opted for buttons, since a) that removes a click from the process, and b) the options should never change from the 4 available selections, and a dropdown kind of implies variable selections.  Only the roles that are actually available for the account will have buttons displayed.

In the jumbotron, instead of just showing the SR for the last played role, I opted to show each available SR based on what roles are currently displayed in the list.  If a role hasn't been played, it will not be displayed in the jumbotron.  If filtering by a specific role, only that role's SR will display in the jumbotron.

Added a column to the games list to denote the role that was played in that game.  This column will only show up when showing all roles.  If you select a single role, it's redundant, so will not be displayed.

The SR Graph page will need some thought put into how it should work with the new system, and how it should look at the transition between the old system and new system.  Showing all 3 roles for all accounts by default could get very busy looking very fast.  The graph itself will probably need some filtering options, and definitely needs some way to differentiate lines, as right now they're all the same purple color.
